pkg/k8s/node_test.go

```diff
@@ -87,6 +87,25 @@ func (s *K8sSuite) TestParseNode(c *C) {
 	c.Assert(n.IPv4AllocCIDR.String(), Equals, "10.254.0.0/16")
 	c.Assert(n.IPv6AllocCIDR, NotNil)
 	c.Assert(n.IPv6AllocCIDR.String(), Equals, "f00d:aaaa:bbbb:cccc:dddd:eeee::/112")
+
+	// No IPv4/IPv6 annotations but PodCIDRs with IPv4/IPv6
+	k8sNode = &slim_corev1.Node{
+		ObjectMeta: slim_metav1.ObjectMeta{
+			Name:        "node2",
+			Annotations: map[string]string{},
+		},
+		Spec: slim_corev1.NodeSpec{
+			PodCIDR:  "10.1.0.0/16",
+			PodCIDRs: []string{"10.1.0.0/16", "f00d:aaaa:bbbb:cccc:dddd:eeee::/112"},
```

Comment:
We test the not-detected case here. Is there a way that we can test the detected case?

Reply:
I added a unit test where the node contains multiple InternalIP/ExternalIP/IPv4/IPv6 addresses, PTAL.